### PR TITLE
add clang-format conf to the project, format the c++ code

### PR DESCRIPTION
**Pull request by adiviness**
## Summary of the Pull Request

- Adds a .clang-format conf file to the project
- Formats all C++ code to be in compliance with the style
- Adds `runformat.cmd` for users to format from the cmd prompt
- Adds `Invoke-CodeFormat` for users to format from powershell
- Adds steps for setting up a custom path to clang-format.exe for Visual Studio users

 
## PR Checklist
* [ ] Closes #xxx
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

### CODE REVIEW

File: tools/OpenConsole.psm1 [hunk 1]
```diff
@@ -238,4 +238,32 @@ function Debug-OpenConsole()
...
+#
+#.PARAMETER Filename
+# The full path to the file to format
+function Invoke-ClangFormat() {
+    param (
+        [parameter(Mandatory=$true)]
+        [string]$Filename
+    )
+    & "$env:OpenconsoleRoot/dep/llvm/clang-format" -i $Filename
+}
```
Comment (DHowett-MSFT) [inline tools/OpenConsole.psm1]: This one is a little bit more robust and supports multiple files:

```powershell
function Invoke-ClangFormat {
    [CmdletBinding()]
    Param (
        [Parameter(Mandatory=$true,ValueFromPipeline=$true)]
        [string[]]$Path
    )

    Process {
        ForEach($_ in $Path) {
            Try {
                $n = Get-Item $_ -ErrorAction Stop | Select -Expand FullName
                & clang-format -i $n
            } Catch {
                Write-Error $_
            }
        }
    }
}
```

just adjust the path to clang-format

then you can make `Invoke-CodeFormat` a lot simpler:

`gci -r "$env:OpenConsoleRoot/src" -include *.cpp,*.hpp,*.h | Where FullName -NotLike "*Generated Files*' | Invoke-ClangFormat`

File: src/interactivity/win32/UiaTextRange.cpp [hunk 2]
```diff
@@ -145,10 +144,10 @@ std::deque<UiaTextRange*> UiaTextRange::GetSelectionRanges(_In_ IRawElementProvi
     return ranges;
 }
 
-
 UiaTextRange* UiaTextRange::Create(_In_ IRawElementProviderSimple* const pProvider)
 {
-    UiaTextRange* range = nullptr;;
+    UiaTextRange* range = nullptr;
+    ;
```
Comment (zadjii-msft) [inline src/interactivity/win32/UiaTextRange.cpp]: hey this one seems wrong

Comment (dlong11) [reply src/interactivity/win32/UiaTextRange.cpp]: The formatting looks right.   The code here looks wrong though.  😄    

I found 5 cases like this in the source. 
src\host\conareainfo.cpp(187):        SMALL_RECT OldRegion = _caInfo.rcViewCaWindow;;
src\host\consoleInformation.cpp(388):    ci.Attributes = GenerateLegacyAttributes(attr);;
src\interactivity\base\InteractivityFactory.cpp(349):    NTSTATUS status = ApiDetector::DetectNtUserWindow(&level);;
src\interactivity\win32\screenInfoUiaProvider.cpp(534):    HRESULT hr = S_OK;;
src\interactivity\win32\UiaTextRange.cpp(151):    UiaTextRange* range = nullptr;;

I can make an issue to clean these up.

Comment (adiviness) [reply src/interactivity/win32/UiaTextRange.cpp]: I can fix them up in this PR, no worries. I already had fixed some of this type of error as part of the initial PR but must've glossed over these.

Comment (DHowett-MSFT) [reply src/interactivity/win32/UiaTextRange.cpp]: I'd rather you not actually -- this pull request should be as close to "whitespace changes only" as possible. Personal preference. What do you think?

Comment (adiviness) [reply src/interactivity/win32/UiaTextRange.cpp]: Sure, I can do it in a separate PR then. I don't have a strong opinion one way or the other.

**Comment by adiviness** [comment]
So I tested it with a sample project and annoyingly the prefast pragmas don't like `\:`. I'll go through and fix them.

**Comment by mikedn** [comment]
Hmm, are you sure you want to put `[[nodiscard]]` on the same line as the declaration? Seems kind of weird and it seems to be responsible for a lot of changes.

**Comment by DHowett-MSFT** [comment]
@mikedn this doesn’t appear to be something that’s configurable with clang-format. It’s important for us to settle on _a_ style, even if we have to make compromises.

If a new version of Clang supports breaking after C++-style attributes, we can revamp our format when we move to it :smile:

**Comment by adiviness** [comment]
I would absolutely prefer if the attributes stayed on their own line but as far as I can tell clang-format doesn't know about them. If it ever does allow styling for them I'll move them back.

File: tools/OpenConsole.psm1 (hunk 1 above)
Comment (ClintyB) [reply tools/OpenConsole.psm1]: [string]$Filename
    )
    & "$env:OpenconsoleRoot/dep/llvm/clang-format" -i $Filename
}

**Comment by DHowett-MSFT** [comment]
@clintyb why

**Comment by adiviness** [comment]
Maybe it's a directory I missed?

**Pull request merged**